IPython/core/interactiveshell.py

```diff
@@ -2674,10 +2674,11 @@ def safe_execfile(self, fname, *where, **kw):
                 # 0
                 # For other exit status, we show the exception unless
                 # explicitly silenced, but only in short form.
-                if kw['raise_exceptions']:
-                    raise
```

Comment:
Yeah. I set `raise_exceptions` to `True` in `shellapp`. If `safe_execfile` doesn't ignore `SystemExit(0)`, an `SystemError` exception will be raised even user's scripts exit normally. 
Actually, I think this is `safe_execfile` intended to do, as it says in comments.
```python
# If the call was made with 0 or None exit status (sys.exit(0)
# or sys.exit() ), don't bother showing a traceback, as both of
# these are considered normal by the OS:
# > python -c'import sys;sys.exit(0)'; echo $?
# 0
# > python -c'import sys;sys.exit()'; echo $?
# 0
# For other exit status, we show the exception unless
# explicitly silenced, but only in short form.
```